### PR TITLE
DOC/CLN: Fix errors in Series docstrings

### PR DESCRIPTION
**Pull request by dsaxton**
Fixes (or starts to fix) a few errors in `Series` docstrings.  On a related note, it seems `validate_docstrings.py` yields this error for a large number of docstrings:

```
The first line of the Returns section should contain only the type, unless multiple values are being returned
```

which seems to happen whenever the return value is named.  Should these be errors?

### CODE REVIEW

**Comment by WillAyd** [comment]
> which seems to happen whenever the return value is named. Should these be errors?

Yea this is intentional. Unless multiple values are being returned should just have a type in the Returns section

File: pandas/core/series.py [hunk 1]
```diff
@@ -3002,7 +3012,7 @@ def argsort(self, axis=0, kind='quicksort', order=None):
 
         Returns
         -------
-        argsorted : Series, with -1 indicated where nan values are present
+        Series, with -1 indicated where nan values are present
```
Comment (WillAyd) [inline pandas/core/series.py]: Split this into a type and description

File: pandas/core/series.py [hunk 2]
```diff
@@ -1394,29 +1399,30 @@ def to_string(self, buf=None, na_rep='NaN', float_format=None, header=True,
...
+            Add the Series name if not None.
         max_rows : int, optional
             Maximum number of rows to show before truncating. If None, show
             all.
 
         Returns
         -------
-        formatted : string (if not buffer passed)
+        str or None
+            String representation of Series if buf=None, otherwise None.
```
Comment (WillAyd) [inline pandas/core/series.py]: You can put code blocks in double backticks for better rendering, so \`\`buf=None\`\`

**Review by WillAyd — CHANGES_REQUESTED** [review]
Minor comments but generally look good. How do the validation scripts look against these?

**Comment by dsaxton** [comment]
This should be the output for the docstrings edited so far in this PR.

`to_string`

```
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
Docstring for "pandas.core.series.Series.to_string" correct. :)
```

`to_dict`

```
2 Warnings found:
	No extended summary found
	See Also section not found
Docstring for "pandas.core.series.Series.to_dict" correct. :)
```

`to_frame`

```
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
Docstring for "pandas.core.series.Series.to_frame" correct. :)
```

`to_sparse`

(I think the first error is wrong here because `"integer"` is actually a string in the docstring.  For this one I'm not familiar enough with the method to add a good description.  Also, I think this will be deprecated soon anyways?)

```
2 Errors found:
	Parameter "kind" type should use "int" instead of "integer"
	Parameter "kind" has no description
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
```

`count`

```
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
Docstring for "pandas.core.series.Series.count" correct. :)
```

`mode`

```
2 Warnings found:
	See Also section not found
	No examples section found
Docstring for "pandas.core.series.Series.mode" correct. :)
```

`drop_duplicates`

(This is another one where the lowercase letter is an argument option.)

```
1 Errors found:
	Parameter "keep" description should start with a capital letter
1 Warnings found:
	No extended summary found
```

`duplicated`

(Same here.)

```
1 Errors found:
	Parameter "keep" description should start with a capital letter
```

`idxmin`

```
2 Errors found:
	Parameters {*args, **kwargs} not documented
	Unknown parameters {*args, **kwargs}
```

`idxmax`

```
2 Errors found:
	Parameters {*args, **kwargs} not documented
	Unknown parameters {*args, **kwargs}
```

`round`

```
1 Errors found:
	Parameters {**kwargs, *args} not documented
2 Warnings found:
	No extended summary found
	No examples section found
```

`quantile`

```
6 Errors found:
	Parameter "q" description should start with a capital letter
	Parameter "q" description should finish with "."
	Parameter "interpolation" has no description
	Return value description should start with a capital letter
	Missing description for See Also "core.window.Rolling.quantile" reference
	Missing description for See Also "numpy.percentile" reference
```

`corr`

```
6 Errors found:
	Parameter "other" has no description
	Parameter "method" description should start with a capital letter
	Parameter "method" description should finish with "."
	Parameter "min_periods" description should finish with "."
	Return value has no description
	flake8 error: E124 closing bracket does not match visual indentation
2 Warnings found:
	No extended summary found
	See Also section not found
```

`cov`

```
4 Errors found:
	Parameter "other" has no description
	Parameter "min_periods" description should finish with "."
	Return value has no description
	Return value has no description
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
```

`diff`

```
Docstring for "pandas.core.series.Series.diff" correct. :)
```

`append`

```
6 Errors found:
	Parameter "to_append" has no description
	Parameter "ignore_index" type should use "bool" instead of "boolean"
	Parameter "verify_integrity" type should use "bool" instead of "boolean"
	Parameter "verify_integrity" description should finish with "."
	Return value has no description
	flake8 error: E231 missing whitespace after ',' (2 times)
1 Warnings found:
	No extended summary found
```

`sort_index`

```
Docstring for "pandas.core.series.Series.sort_index" correct. :)
```

`argsort`

```
4 Errors found:
	Summary must start with infinitive verb, not third person (e.g. use "Generate" instead of "Generates")
	Summary should fit in a single line
	Parameter "kind" description should finish with "."
	Missing description for See Also "numpy.ndarray.argsort" reference
1 Warnings found:
	No examples section found
```

`swaplevel`

```
4 Errors found:
	Parameters {j, copy, i} not documented
	Unknown parameters {i, j}
	Parameter "i, j" type should use "str" instead of "string"
	Return value has no description
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
```

`unstack`

```
7 Errors found:
	Summary should fit in a single line
	Parameter "level" type should use "str" instead of "string"
	Parameter "level" description should finish with "."
	Parameter "fill_value" description should start with a capital letter
	Parameter "fill_value" description should finish with "."
	Return value has no description
	flake8 error: E128 continuation line under-indented for visual indent
1 Warnings found:
	See Also section not found
```

`rename`

```
7 Errors found:
	Parameters {**kwargs} not documented
	Unknown parameters {level, inplace, copy}
	Parameter "index" description should start with a capital letter
	Parameter "copy" description should finish with "."
	Return value has no description
	Missing description for See Also "Series.rename_axis" reference
	flake8 error: E261 at least two spaces before inline comment
```

`drop`

```
4 Errors found:
	Parameters {columns, index} not documented
	Unknown parameters {index, columns}
	Return value has no description
	flake8 error: E231 missing whitespace after ',' (3 times)
```

`isin`

```
Docstring for "pandas.core.series.Series.isin" correct. :)
```

`between`

```
Docstring for "pandas.core.series.Series.between" correct. :)
```

`from_csv`

```
3 Errors found:
	Parameter "path" has no description
	Return value has no description
	Missing description for See Also "read_csv" reference
1 Warnings found:
	No examples section found
```

`valid`

```
2 Errors found:
	Parameters {inplace, **kwargs} not documented
	No Returns section found
2 Warnings found:
	See Also section not found
	No examples section found
```

`to_period`

```
2 Errors found:
	Summary should fit in a single line
	Parameters {copy} not documented
2 Warnings found:
	See Also section not found
	No examples section found
```

**Comment by dsaxton** [comment]
@WillAyd Side question, but when I merge with master on GitHub, how do I prevent lots of unrelated files from being marked as edited in my PR when I pull locally and then later push?

**Comment by WillAyd** [comment]
Typically I would do the below locally on branch in question

```sh
git fetch upstream
git merge upstream/master
<FIX CONFLICTS IF ANY, then git merge --continue>
git push origin your-branch-name
```

**Comment by jreback** [comment]
can you merge master

**Comment by WillAyd** [comment]
@dsaxton seeing some movement on this and maybe other PRs. Please ping when you are ready for review

**Comment by dsaxton** [comment]
@WillAyd Would be happy to have you take a look whenever you get a chance

**Comment by WillAyd** [comment]
Thanks - latest version looks like a good improvement over what was there previously. Can you provide updated script validations? Probably OK to leave a few out there for follow up PRs just want to make sure we haven't increased number of errors before approving

**Comment by dsaxton** [comment]
@WillAyd Still some errors / warnings here and there (some of which I don't think are real, others I'm not certain of the fix), but I think we've cleaned up a good chunk of them.

```
to_string

3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found
Docstring for "pandas.core.series.Series.to_string" correct. :)

to_dict

2 Warnings found:
	No extended summary found
	See Also section not found
Docstring for "pandas.core.series.Series.to_dict" correct. :)

to_frame

2 Warnings found:
	No extended summary found
	See Also section not found
Docstring for "pandas.core.series.Series.to_frame" correct. :)

to_sparse

2 Errors found:
	Parameter "kind" type should use "int" instead of "integer"
	Parameter "kind" has no description
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found

count

2 Warnings found:
	No extended summary found
	See Also section not found
Docstring for "pandas.core.series.Series.count" correct. :)

mode

2 Warnings found:
	See Also section not found
	No examples section found
Docstring for "pandas.core.series.Series.mode" correct. :)

drop_duplicates

1 Errors found:
	Parameter "keep" description should start with a capital letter
1 Warnings found:
	No extended summary found

duplicated

1 Errors found:
	Parameter "keep" description should start with a capital letter

idxmin

2 Errors found:
	Parameters {**kwargs, *args} not documented
	Unknown parameters {*args, **kwargs}

idxmax

2 Errors found:
	Parameters {**kwargs, *args} not documented
	Unknown parameters {*args, **kwargs}

round

1 Errors found:
	Parameters {*args, **kwargs} not documented
1 Warnings found:
	No extended summary found

quantile

4 Errors found:
	Parameter "q" description should start with a capital letter
	Parameter "interpolation" has no description
	Missing description for See Also "core.window.Rolling.quantile" reference
	Missing description for See Also "numpy.percentile" reference
1 Warnings found:
	No extended summary found

corr

(flake8 is complaining about this code block, which I don't think is necessarily wrong.)

        >>> def histogram_intersection(a, b):
        ...     v = np.minimum(a, b).sum().round(decimals=1)
        ...     return v
        >>> s1 = pd.Series([.2, .0, .6, .2])
        >>> s2 = pd.Series([.3, .6, .0, .1])
        >>> s1.corr(s2, method=histogram_intersection)
        0.3

4 Errors found:
	Parameter "method" description should start with a capital letter
	Parameter "method" description should finish with "."
	flake8 error: E302 expected 2 blank lines, found 0
	flake8 error: E305 expected 2 blank lines after class or function definition, found 0
2 Warnings found:
	No extended summary found
	See Also section not found

cov

2 Warnings found:
	No extended summary found
	See Also section not found
Docstring for "pandas.core.series.Series.cov" correct. :)

diff

Docstring for "pandas.core.series.Series.diff" correct. :)

append

1 Warnings found:
	No extended summary found
Docstring for "pandas.core.series.Series.append" correct. :)

sort_index

Docstring for "pandas.core.series.Series.sort_index" correct. :)

argsort

4 Errors found:
	Summary must start with infinitive verb, not third person (e.g. use "Generate" instead of "Generates")
	Summary should fit in a single line
	Parameter "kind" description should finish with "."
	Missing description for See Also "numpy.ndarray.argsort" reference
1 Warnings found:
	No examples section found

swaplevel

2 Errors found:
	Parameters {i, j, copy} not documented
	Unknown parameters {i, j}
3 Warnings found:
	No extended summary found
	See Also section not found
	No examples section found

unstack

1 Errors found:
	Summary should fit in a single line
1 Warnings found:
	See Also section not found

rename

3 Errors found:
	Parameters {**kwargs} not documented
	Unknown parameters {level, copy, inplace}
	Parameter "index" description should start with a capital letter

drop

2 Errors found:
	Parameters {columns, index} not documented
	Unknown parameters {index, columns}

isin

Docstring for "pandas.core.series.Series.isin" correct. :)

between

Docstring for "pandas.core.series.Series.between" correct. :)

from_csv

3 Errors found:
	Parameter "path" has no description
	Return value has no description
	Missing description for See Also "read_csv" reference
1 Warnings found:
	No examples section found

valid

2 Errors found:
	Parameters {**kwargs, inplace} not documented
	No Returns section found
2 Warnings found:
	See Also section not found
	No examples section found

to_period

1 Errors found:
	Summary should fit in a single line
2 Warnings found:
	See Also section not found
	No examples section found
```

**Comment by WillAyd** [comment]
Makes sense. I checked most remaining and most are errors with the validator itself (ex: things that start with bullet points in description). I think this is a good enough improvement to move forward so we don't get stuck in a holding pattern forever.

cc @datapythonista - looks like there might be a new error for the below line, where `'integer'` is a valid argument but the validator complains about it not being written out as `int`. I'll open up an issue for that separately

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @dsaxton !

**Pull request merged**